src/tools/dtrust-tool.c

```diff
@@ -196,6 +196,7 @@ unlock_transport_protection(sc_card_t *card)
 	else
 		printf("Can't change pin: %s\n", sc_strerror(r));
 
+fail_qespincmp:
```

Comment:
I am not a huge fan of the huge amount of `goto`/`labels` for different stages of fail. The code looks ok, but I would prefer to have only one label `fail`.
Given that the variables are always compared with NULL and they are initialized to NULL when declared it is safe to run anything from here below at any failure case, isn't it?

So my preference would be to define here only one label `fail`, remove the others and change all `goto`s to `fail`. I think that should not make any difference in behavior, but the code would be much easier to understand.